core/src/test/java/jenkins/util/io/PathRemoverTest.java

```diff
@@ -418,6 +424,34 @@ public void testForceRemoveRecursive_ParentIsSymbolicLink() throws IOException {
         assertFalse(d1.exists());
     }
 
+    @Test
+    @Issue("JENKINS-55448")
+    public void testForceRemoveRecursive_TruncatesNumberOfExceptions() throws IOException {
+        assumeTrue(Functions.isWindows());
+        final int maxExceptions = CompositeIOException.EXCEPTION_LIMIT;
+        final int lockedFiles = maxExceptions + 5;
+        final int totalFiles = lockedFiles + 5;
+        File dir = tmp.newFolder();
+        File[] files = new File[totalFiles];
+        for (int i = 0; i < totalFiles; i++) {
+            files[i] = new File(dir, "f" + i);
+        }
+        touchWithFileName(files);
+        for (int i = 0; i < lockedFiles; i++) {
+            locker.acquireLock(files[i]);
+        }
+        try {
+            PathRemover.newSimpleRemover().forceRemoveRecursive(dir.toPath());
+            fail("Deletion should have failed");
+        } catch (CompositeIOException e) {
+            assertThat(e.getSuppressed(), arrayWithSize(maxExceptions + 1));
```

Comment:
Notably in this case, the suppressed exceptions in the outer `CompositeIOException` are actually instances of `CompositeIOException` themselves because of the fallback behavior in `PathRemover.removeOrMakeRemovableThenRemove`, each with two suppressed instances of `FileSystemException`. That means that the worst-case number of exceptions in a single `CompositeIOException` for the deletion operations in `PathRemover` with this patch is actually `CompositeIOException.EXCEPTION_LIMIT * 3`.